C7/PCXToGodot.cs

```diff
@@ -95,21 +95,48 @@ private static Image ByteArrayToImage(byte[] colorIndices, byte[,] palette, int
 		return getImageFromBufferData(width, height, BufferData);
 	}
 
+	// ByteArrayWithTintToImage is used to load create images from flic frames
+	// that contain a tinted layer such as unit animations, where the unit's
+	// clothing is tinted by their civ color.
+	public static (Image, Image) ByteArrayWithTintToImage(byte[] colorIndices, byte[,] palette, int width, int height, int[] transparent = null, bool shadows = false) {
+		int[] colorData = loadPalette(palette, shadows);
+		int[] baseLayer = new int[width * height];
+		int[] tintLayer = new int[width * height];
+
+		Pcx whitePcx = Util.LoadPCX("Art/Units/Palettes/ntp00.pcx");
+		int[] whiteColorData = loadPalette(whitePcx.Palette, true);
+
+		for (int i = 0; i < width * height; i++) {
+			int index = colorIndices[i];
+			bool tinted = index < 16 || (index < 64 && index % 2 == 0);
+			bool shadow = index >= 224 && index <= 239;
+			if (tinted) {
+				tintLayer[i] = whiteColorData[index];
+				baseLayer[i] = 0; // transparent
+			} else if (shadow) {
+				// shadow belongs to the base texture
+				baseLayer[i] = ((int)new Color(1.0f, 1.0f, 1.0f, (float)index - 224f / 239f - 224f).ToArgb32());
```

Comment:
there's extra float math here that could be simplified to (float)(index-224)/15

also more constants for readability